app/controllers/api/encryptables_controller.rb

```diff
@@ -165,19 +169,29 @@ def fetch_encryptable_files
   def build_encryptable_file
     filename = params[:file].original_filename
 
-    file = new_file(file_credential, params[:description], filename)
+    file = if credential_id.present?
+             new_attachment(credential, params[:description], filename)
+           else
+             new_file(folder, params[:description], filename)
+           end
     file.content_type = params[:file].content_type
     file.cleartext_file = params[:file].read
 
     instance_variable_set(:"@#{ivar_name}", file)
   end
 
-  def new_file(parent_encryptable, description, name)
+  def new_attachment(parent_encryptable, description, name)
```

Comment:
I would rather find a way to define the correct model params and use dry crud functionality for creating the file entry.